homeassistant/components/overkiz/water_heater_entities/domestic_hot_water_production.py

```diff
@@ -58,17 +55,21 @@ def __init__(
         super().__init__(device_url, coordinator)
 
         # Init operation mode to set for this specific device
-        self.overkiz_to_operation_mode: dict[str, str] = {}
+        self.operation_mode_to_overkiz: dict[str, str] = {}
+        self._attr_operation_list = []
         state_mode_definition = self.executor.select_definition_state(
             OverkizState.IO_DHW_MODE, OverkizState.MODBUSLINK_DHW_MODE
         )
-        if state_mode_definition and state_mode_definition.values:
+        for param, mode in OVERKIZ_TO_OPERATION_MODE.items():
             # Filter only for mode allowed by this device
-            for param, mode in OVERKIZ_TO_OPERATION_MODE.items():
-                if param in state_mode_definition.values:
-                    self.overkiz_to_operation_mode[param] = mode
-        else:
-            self.overkiz_to_operation_mode = OVERKIZ_TO_OPERATION_MODE
+            # or allow all if no mode definition found
+            if (
+                not state_mode_definition
+                or state_mode_definition.values
+                and param in state_mode_definition.values
+            ):
+                self.operation_mode_to_overkiz[mode] = param
+                self._attr_operation_list.append(param)
```

Comment:
Would it make more sense to pull this logic out to the `operation_list` property instead of dynamically setting the internal static attribute on initialization? That's how I've seen it done in other entities which have a property that changes based on the device.

Reply:
This entity handle more than 10 different devices using the same UIWidget in overkiz.
They share some logics, but they don't provide the exact same functionnalities not the same DHW modes.  
This code is here to detect exaclty which DHW modes is supported by the real devices which tell us what value we could use.

Other overkiz entities usually works for 1 physical device, that's why the code is simpler in other places.

Reply:
@kevin-brown I'm not sure to understand how I'm supposed to do that.  
Could you point me to an exemple in the code please?

Reply:
Since the operation list is constant after initialization we prefer using an instance attribute over a property.

Reply:
> Since the operation list is constant after initialization we prefer using an instance attribute over a property.

That's fair, I can get behind that differentiation here.